connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStoreTest.java

```diff
@@ -105,34 +112,53 @@ public class KafkaOffsetBackingStoreTest {
 
     @Mock
     KafkaBasedLog<byte[], byte[]> storeLog;
+    @Mock
+    Supplier<TopicAdmin> topicAdminSupplier;
+    @Spy
+    @InjectMocks
     private KafkaOffsetBackingStore store;
 
-    private Capture<String> capturedTopic = EasyMock.newCapture();
-    private Capture<Map<String, Object>> capturedProducerProps = EasyMock.newCapture();
-    private Capture<Map<String, Object>> capturedConsumerProps = EasyMock.newCapture();
-    private Capture<Supplier<TopicAdmin>> capturedAdminSupplier = EasyMock.newCapture();
-    private Capture<NewTopic> capturedNewTopic = EasyMock.newCapture();
-    private Capture<Callback<ConsumerRecord<byte[], byte[]>>> capturedConsumedCallback = EasyMock.newCapture();
+    private MockedStatic<ConnectUtils> connectUtilsMockedStatic;
+
+    private final ArgumentCaptor<String> capturedTopic = ArgumentCaptor.forClass(String.class);
+    @SuppressWarnings("unchecked")
+    private final ArgumentCaptor<Map<String, Object>> capturedProducerProps = ArgumentCaptor.forClass(Map.class);
+    @SuppressWarnings("unchecked")
+    private final ArgumentCaptor<Map<String, Object>> capturedConsumerProps = ArgumentCaptor.forClass(Map.class);
+    @SuppressWarnings("unchecked")
+    private final ArgumentCaptor<Supplier<TopicAdmin>> capturedAdminSupplier = ArgumentCaptor.forClass(Supplier.class);
+    private final ArgumentCaptor<NewTopic> capturedNewTopic = ArgumentCaptor.forClass(NewTopic.class);
+    @SuppressWarnings("unchecked")
+    private final ArgumentCaptor<Callback<ConsumerRecord<byte[], byte[]>>> capturedConsumedCallback = ArgumentCaptor.forClass(Callback.class);
 
     @Before
-    public void setUp() throws Exception {
-        store = PowerMock.createPartialMockAndInvokeDefaultConstructor(KafkaOffsetBackingStore.class, "createKafkaBasedLog");
-    }
+    public void setup() {
+        doReturn(storeLog).when(store).createKafkaBasedLog(capturedTopic.capture(), capturedProducerProps.capture(),
+                capturedConsumerProps.capture(), capturedConsumedCallback.capture(),
+                capturedNewTopic.capture(), capturedAdminSupplier.capture());
 
-    @Test
-    public void testStartStop() throws Exception {
-        expectConfigure();
-        expectStart(Collections.emptyList());
-        expectStop();
-        expectClusterId();
+        connectUtilsMockedStatic = mockStatic(ConnectUtils.class, Mockito.CALLS_REAL_METHODS);
+        connectUtilsMockedStatic.when(() -> ConnectUtils.lookupKafkaClusterId(any())).thenReturn("test-cluster");
+    }
 
-        PowerMock.replayAll();
+    @After
+    public void tearDown() {
+        connectUtilsMockedStatic.close();
```

Comment:
We should also make sure that none of the tests resulted in unexpected calls to start, stop, etc. the underlying `KafkaBasedLog`:
```suggestion
        connectUtilsMockedStatic.close();
        Mockito.verifyNoMoreInteractions(storeLog);
```

Reply:
Added the additional verification to the next commit.